boost.cmake

```diff
@@ -41,11 +41,53 @@ MACRO(SEARCH_FOR_BOOST)
     SET(BOOST_REQUIRED 1.40)
   ENDIF(NOT BOOST_REQUIRED)
 
+  # First try to find Boost to get the version
+  FIND_PACKAGE(Boost ${BOOST_REQUIRED})
+
   IF(NOT DEFINED BOOST_COMPONENTS)
     SET(BOOST_COMPONENTS
       filesystem system thread program_options unit_test_framework)
   ENDIF(NOT DEFINED BOOST_COMPONENTS)
 
+  # Check if python is in the list and adjust the version according to the current Python version.
+  # This is made mandatory if for Boost version greater than 1.67.0
+  IF(Boost_VERSION VERSION_GREATER "1.67.0")
+    LIST(FIND BOOST_COMPONENTS python PYTHON_IN_BOOST_COMPONENTS)
+    SET(BOOST_COMPONENTS_ ${BOOST_COMPONENTS})
+    IF(${PYTHON_IN_BOOST_COMPONENTS} GREATER -1)
+      # Check if Python has been found
+      IF(NOT PYTHONLIBS_FOUND)
+        MESSAGE(FATAL_ERROR "PYTHON has not been found. Please first check for Python.")   
```

Comment:
So the intent here is to imply that `FindPython` is called before `Search_For_Boost` if the former contains the python component right?

I think the message could contain this as a hint (as this is potentially breaking existing scripts)

Reply:
Yes, indeed.
I will do also the modification.